.github/workflows/main.yml

```diff
@@ -57,37 +58,57 @@ jobs:
         else
           echo "::set-output name=result::odd"
         fi
+  build-and-cache-node-modules:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Setup repo
+        uses: actions/checkout@v3
+         
+      - uses: actions/setup-node@v3
+        with: 
+          registry-url: ${{ env.NPM_REGISTRY }}
+          scope: "${{ env.SCOPE }}"
+          cache: 'npm'
+
+      - name: Setup Deno
+        uses: denoland/setup-deno@v1
+        with:
+          deno-version: "1.23.4"
+
+      - name: Build all
+        run: |
+          export PATH="$PATH:$(pwd)"
```

Comment:
I think the preferred way to get things in the path is through appending it to the $GITHUB_PATH env var like they do here: https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#adding-a-system-path